File: thesis/src/sources/echo/context_0_1.py
import values
import sys
value_file = open("fpCH2nSTj_values.txt", "w")
exception_file = open("fpCH2nSTj_exceptions.txt", "w")
def write_separator():
    value_file.write("--fpCH2nSTj-- SEP")
    exception_file.write("--fpCH2nSTj-- SEP")
    sys.stderr.write("--fpCH2nSTj-- SEP")
    sys.stdout.write("--fpCH2nSTj-- SEP")
    sys.stdout.flush()
    sys.stderr.flush()
    value_file.flush()
    exception_file.flush()
def send_value(value):
    values.send_value(value_file, value)
def send_exception(exception):
    values.send_exception(exception_file, exception)
def send_specific_value(r):
    values.send_evaluated(value_file, r)
def send_specific_exception(r):
    values.send_evaluated(exception_file, r)
try:
    write_separator()
    from submission import *
except Exception as e:
    send_exception(e)
else:
    send_exception(None)
value_file.close()
exception_file.close()

File: thesis/src/sources/echo/context01.js
const fs = require('fs');
const vm = require('vm');
const values = require("./values.js");
const valueFile = fs.openSync("TMIQgB2wo_values.txt", "w");
const exceptionFile = fs.openSync("TMIQgB2wo_exceptions.txt", "w");
function writeSeparator() {
    fs.writeSync(valueFile, "--TMIQgB2wo-- SEP");
    fs.writeSync(exceptionFile, "--TMIQgB2wo-- SEP");
    fs.writeSync(process.stdout.fd, "--TMIQgB2wo-- SEP");
    fs.writeSync(process.stderr.fd, "--TMIQgB2wo-- SEP");
}
function sendValue(value) {
    values.sendValue(valueFile, value);
}
function sendException(exception) {
    values.sendException(exceptionFile, exception);
}
function sendSpecificValue(value) {
    values.sendEvaluated(valueFile, value);
}
function sendSpecificException(exception) {
    values.sendEvaluated(exceptionFile, exception);
}
new_args = [process.argv[0]]
new_args = new_args.concat([])
process.argv = new_args
try {
    writeSeparator();
    eval(fs.readFileSync("submission.js") + "");
    sendException(null)
} catch(e) {
    sendException(e)
}
fs.closeSync(valueFile);
fs.closeSync(exceptionFile);